app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageComposeInputState.kt

```diff
@@ -100,10 +101,12 @@ sealed class MessageComposeInputType {
 
     @Stable
     data class SelfDeletingMessage(
-        val selfDeletionDuration: SelfDeletionDuration
+        val selfDeletionDuration: SelfDeletionDuration,
+        val attachmentOptionsDisplayed: Boolean = false,
```

Comment:
Why don't we extract this to the `MessageComposeInputState` level instead of passing through the same value to `NewMessage` and `SelfDeletingMessage` types? Making it as another abstract value that every sealed subclass overrides, or simply declaring it as another mutable variable on the `MessageComposerState` class would make more sense to me.

Reply:
I followed what was already done previously NewMessage and EditMessage are the types for the input it would be weird to do something else at this point in my opinion, we would have two ways of changing the compose input type. 

also editing messages should not care about selfDeletionDuration also NewMessage type should not care about this as well once we are into this input type.

Reply:
Also the approoach here is something I would like to take into the refactor and I think kind of makes sense, so haaving this done like this would make the refactor abit easier.

Reply:
I meant the `attachmentOptionsDisplayed` not the self deleting duration 🙃

Reply:
but that would imply EditMessageInput to care about attachmentOptionsDisplayed 🤔

Reply:
however I see what you mean and now I read through the code I think you got the point, I am going to add a TODO to take another look at it, as I am scared of doing anything to message composer at the moment xd and this would require some other changes

Reply:
Ok. Let's skip this discussion for now as it will be tackled in a upcoming refactoring PR.